lit.ts

```diff
@@ -644,54 +644,83 @@ export async function queryCapacityCredits(signer: ethers.Wallet) {
 }
 
 export async function addPaymentDelegationPayee({
-	wallet,
-	payeeAddresses,
+    wallet,
+    payeeAddresses,
 }: {
-	wallet: ethers.Wallet;
-	payeeAddresses: string[];
+    wallet: ethers.Wallet;
+    payeeAddresses: string[];
 }) {
-	// get the first token that is not expired
-	const capacityTokens: CapacityToken[] = await queryCapacityCredits(wallet);
-	console.log("Got capacity tokens", JSON.stringify(capacityTokens, null, 2));
-	const capacityToken = capacityTokens.find((token) => !token.isExpired);
-
-	let tokenId: number | null = null;
-
-	if (!capacityToken) {
-		// mint a new token
-		const minted = await mintCapacityCredits({ signer: wallet });
-
-		if (!minted) {
-			throw new Error("Failed to mint capacity credits");
-		}
-
-		console.log(
-			"No capacity token found, minted a new one:",
-			minted.capacityTokenId,
-		);
-		tokenId = minted.capacityTokenId;
-	} else {
-		tokenId = capacityToken.tokenId;
-	}
-
-	if (!tokenId) {
-		throw new Error("Failed to get ID for capacity token");
-	}
-
-	// add payer in contract
-	const provider = getProvider();
-	const paymentDelegationContract = await getContractFromJsSdk(
-		config.network,
-		"PaymentDelegation",
-		wallet,
-	);
-
-	const tx = await paymentDelegationContract.functions.delegatePaymentsBatch(
-		payeeAddresses,
-	);
-	console.log("tx hash for delegatePaymentsBatch()", tx.hash);
-	await tx.wait();
-	return tx;
+    // get the first token that is not expired
+    const capacityTokens: CapacityToken[] = await queryCapacityCredits(wallet);
+    console.log("Got capacity tokens", JSON.stringify(capacityTokens, null, 2));
+    const capacityToken = capacityTokens.find((token) => !token.isExpired);
+
+    let tokenId: number | null = null;
+
+    if (!capacityToken) {
+        // mint a new token
+        const minted = await mintCapacityCredits({ signer: wallet });
+
+        if (!minted) {
+            throw new Error("Failed to mint capacity credits");
+        }
+
+        console.log(
+            "No capacity token found, minted a new one:",
+            minted.capacityTokenId,
+        );
+        tokenId = minted.capacityTokenId;
+    } else {
+        tokenId = capacityToken.tokenId;
+    }
+
+    if (!tokenId) {
+        throw new Error("Failed to get ID for capacity token");
+    }
+
+    // add payer in contract
+    const provider = getProvider();
+    const paymentDelegationContract = await getContractFromJsSdk(
+        config.network,
+        "PaymentDelegation",
+        wallet,
+    );
+
+    // First try without manual gas limit
```

Comment:
let's just use the gas estimate for every txn and not do a retry type thing.  and let's use math that works with BigNumber, since the gas limit will be a BigNumber.  there's an example of how to do this math here: https://github.com/LIT-Protocol/relay-server/blob/main/lit.ts#L259